prestodb/dbapi.py

```diff
@@ -74,23 +75,43 @@ class Connection(object):
     def __init__(self, *args, **kwargs):
         self._args = args
         self._kwargs = kwargs
+        self._transaction = None
+        self.isolation_level = kwargs.pop('isolation_level', None)
 
     def close(self):
         """Presto does not have anything to close"""
         # TODO cancel outstanding queries?
         pass
 
+    def start_transaction(self):
+        request = prestodb.client.PrestoRequest(*self._args, **self._kwargs)
+        request.transaction_id = 'NONE'
+        self._transaction = Transaction(request)
+        self._transaction.begin()
+        return self._transaction
+
     def commit(self):
```

Comment:
How about calling this `commit_transaction` to pair with the `start_transaction`

Reply:
[DBAPI 2.0](https://www.python.org/dev/peps/pep-0249/#connection-methods) defines an interface with `.commit()`. However there is no `.start_transaction()`.  It can be seen as a convenience method and extension of the API. I could make rename it to `._start_transaction()`.

If the implementation of `Connection` follows a RAII-like pattern, it should start the transaction at instantiation time when `isolation_level` is non-zero.  Here the transaction is bound to `Connection` and not cursor.

Reply:
To give more context, `isolation_level` is not defined by DBAPI 2.0. I took inspiration from [sqlite3](https://docs.python.org/3/library/sqlite3.html). The main problem I tried to solve with this attribute was to toggle on/off the *autocommit* behavior.

Reply:
Isolation level is also passed to the [`START TRANSACTION`](https://prestodb.io/docs/current/sql/start-transaction.html) statement. I will:
- Keep the `isolation_level` attribute and make it a read-only property
- Start the transaction in `Connection`'s constructor when `isolation_level` is non-zero
- Provide a constant `AUTOCOMMIT=0`, so `Connection` could be instantiated with `Connection(isolation_level=AUTOCOMMIT, ...)`
- Provide constants for all isolation level defined by Presto

Reply:
(2.) can't work because we use the same `Connection` object to create multiple `Cursor`s. So the transaction really has to start in `Connection.cursor()`.